File: frontend/packages/api/src/__generated__/types/StackEnvironmentVariableDto.ts
import type { EnvironmentVariableType } from './EnvironmentVariableType'

export type StackEnvironmentVariableDto = {
  /**
   * @type string, uuid
   */
  id: string
  /**
   * @type string, uuid
   */
  stackId: string
  /**
   * @type string, uuid
   */
  groupId: string | null
  /**
   * @type string
   */
  key: string
  /**
   * @type string
   */
  value: string
  /**
   * @type boolean
   */
  isSecret: boolean
  /**
   * @type integer
   */
  type: EnvironmentVariableType
  /**
   * @type string, date-time
   */
  createdAt: string
  /**
   * @type string, date-time
   */
  lastModifiedAt?: string | null
  /**
   * @type string
   */
  groupName?: string | null
}

File: backend/src/CloudCrafter.Domain/Domain/Stacks/StackEnvironmentVariableDto.cs
using AutoMapper;
using CloudCrafter.Domain.Common.AutoMapper;
using CloudCrafter.Domain.Entities;

namespace CloudCrafter.Domain.Domain.Stacks;

public class StackEnvironmentVariableDto
{
    public required Guid Id { get; init; }
    public required Guid StackId { get; init; }
    public required Guid? GroupId { get; init; }
    public required string Key { get; init; }
    public required string Value { get; init; }
    public required bool IsSecret { get; init; }
    public required EnvironmentVariableType Type { get; init; }
    public required DateTime CreatedAt { get; init; }
    public DateTime? LastModifiedAt { get; init; }
    public string? GroupName { get; init; }

    private class Mapping : Profile
    {
        public Mapping()
        {
            CreateMap<StackEnvironmentVariable, StackEnvironmentVariableDto>()
                .ForMember(dest => dest.Value, opt => opt.MapFrom<SecretValueResolver>());
        }
    }
}
